packages/core-data/src/entities.js

```diff
@@ -449,15 +473,17 @@ function registerSyncConfigs( configs ) {
 /**
  * Loads the kind entities into the store.
  *
- * @param {string} kind Kind
- *
+ * @param {string}  kind Kind
+ * @param {?string} name Name
  * @return {(thunkArgs: object) => Promise<Array>} Entities
  */
 export const getOrLoadEntitiesConfig =
-	( kind ) =>
+	( kind, name ) =>
```

Comment:
Why does this function need a "name" now. I believe the purpose of this function is to load the entities of a given "kind", so "name" is not needed?

Reply:
The `root` is a standard `kind`, and execution never reaches additional loaders. I'm using the `name` to narrow things down.

Reply:
I admin that I'm not following. There's something I'm obviously missing.

Reply:
* Every entity in `rootEntitiesConfig` has a `kind` root (as the const name suggests).
* The `select.getEntitiesConfig( kind )` will always return the list of configs for `kind === root`
* The condition in the code below is met, and configs objects are returned.
* The `getOrLoadEntitiesConfig` never reaches the addition entity config loading logic to load the `site` entity.

https://github.com/WordPress/gutenberg/blob/ec86a82ca6aa66887c2509e88ce0a584f759c398/packages/core-data/src/entities.js#L459-L468

Reply:
Can we instead update the condition to check whether a key exist in `additionalEntityConfigLoaders` and maybe have some "meta" state to indicate that we're loading or we've already loaded (maybe we can even reuse the metadata state that exist in all stores)

Reply:
I've been thinking more, basically what we want is a resolver for. `getEntityConfig( kind, name )`. In other words, I'm ok with passing the name but we at least have to rename the function.

Reply:
Sure, I'm happy to give it a try. However, we might encounter a similar issue without using the entity `name` when `additionalEntityConfigLoaders` contains more than one entity of the same `kind`.

Reply:
Let's keep your changes but just clarify things in the description. I'm not entirely satisfied with how we're dealing with all of that, but it's not clear to me what's the best path forward is yet:

 - We have static entities (kind, name)
 - We have dynamic entities: For these entities, for some the kind is enough to resolve their config, for others we need both the kind and name. 

It seems there's potentially a better way/API to "register" these entities.